### DISCUSSION TITLE
initial setup issue in backend-core

### DISCUSSION THREAD

**Discussion by freekvdp** [Budibase/budibase#12263] [Help]
I'm trying to build the repo for the first time but can't seem to get past this error:

`@budibase/server: node:internal/modules/cjs/loader:494
@budibase/server:       throw e;
@budibase/server:       ^
@budibase/server: Error: Cannot find module '/Users/freekvdp/Projects/budibase/node_modules/@budibase/backend-core/dist/src/db.js'
@budibase/server:     at createEsmNotFoundErr (node:internal/modules/cjs/loader:967:15)
@budibase/server:     at finalizeEsmResolution (node:internal/modules/cjs/loader:960:15)
@budibase/server:     at resolveExports (node:internal/modules/cjs/loader:488:14)
@budibase/server:     at Module._findPath (node:internal/modules/cjs/loader:528:31)
@budibase/server:     at Module._resolveFilename (node:internal/modules/cjs/loader:932:27)
@budibase/server:     at Module._load (node:internal/modules/cjs/loader:787:27)
@budibase/server:     at Module.require (node:internal/modules/cjs/loader:1012:19)
@budibase/server:     at require (node:internal/modules/cjs/helpers:102:18)
@budibase/server:     at src/sdk/app/tables/migration.ts (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:26598:17)
@budibase/server:     at __init (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:9:59)
@budibase/server:     at src/sdk/app/tables/index.ts (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:26667:5)
@budibase/server:     at __init (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:9:59)
@budibase/server:     at src/sdk/index.ts (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:30421:5)
@budibase/server:     at __init (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:9:59)
@budibase/server:     at src/db/linkedRows/index.ts (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:30607:5)
@budibase/server:     at __init (/Users/freekvdp/Projects/budibase/packages/server/dist/index.js:9:59) {
@budibase/server:   code: 'MODULE_NOT_FOUND',
@budibase/server:   path: '/Users/freekvdp/Projects/budibase/node_modules/@budibase/backend-core/package.json'
@budibase/server: }
@budibase/server: Node.js v18.4.0`

The location of the module is indeed not at "backend-core/dist/src/db.js" but should probably be "backend-core/dist/src/db/db.js". I tried building with with different node-versions, but that didn't change anything.
Now I'm stuck and have no clue how/where the path should be updated...

I work on a mbp m1.

**Comment by nadeem55** [comment]
Any luck with this issue?

**Reply by freekvdp** [reply]
unfortunately not...

**Reply by freekvdp** [reply]
So I found the issue and solved it by adding '/index' to 2 import paths. But then I noticed I was building master and never tried to build the develop branch. On develop branch the issue is nonexistent.

**Reply by mikeluigijean** [reply]
where did you add '/index' to the 2 import paths?

**Reply by kuldeepsmane** [reply]
I also did not get where to add '/index'. @freekvdp

**Reply by freekvdp** [reply]
<img width="511" alt="image" src="https://github.com/Budibase/budibase/assets/22095093/a7879b01-c307-431e-a617-e4b964a4156a">
<img width="453" alt="image" src="https://github.com/Budibase/budibase/assets/22095093/7f0b57c9-6240-4d4f-9f5e-d33d8e8fccd0">
